city_scrapers/spiders/mi_belle_isle.py

```diff
@@ -0,0 +1,108 @@
+                'end': {
+                    'date': date,
+                    'time': end_time,
+                    'note': '',
+                },
+                'all_day': False,
+                'location': self._parse_location(item),
+                'documents': self._match_documents(item, response),
+                'sources': [{'url': response.url, 'note': ''}],
+            }
+
+            data['status'] = self._generate_status(data, text='')
+            data['id'] = self._generate_id(data)
+
+            yield data
+
+    def _parse_date_and_times(self, item):
+        """
+        Parse start and end date and times.
+        """
+        date_str = item.xpath('.//td[1]/text()').re(r'[^\*]+')[0]
+        time_str = item.xpath('.//td[2]/text()').extract_first().replace('.', '')
+        meridian_str = re.findall(r'am|pm', time_str.lower())[0]
+
+        time_start_str = re.findall(r'(\d+?:*?\d*?)(?=\s*-)', time_str)[0]
+        time_end_str = re.findall(r'((?<=-)\s*)(\d+?:*?\d*)', time_str)[0][1]
+
+        date_value = dateparse(date_str)
+        start_value = dateparse('{0} {1} {2}'.format(date_str, time_start_str, meridian_str))
+        end_value = dateparse('{0} {1} {2}'.format(date_str, time_end_str, meridian_str))
+
+        return date_value.date(), start_value.time(), end_value.time()
+
+    def _parse_location(self, item):
+        """
+        Parse or generate location. Latitude and longitude can be
+        left blank and will be geocoded later.
+        """
+        location_name = item.xpath('.//td[3]/text()').extract_first()
+        if "flynn" in location_name.lower():
+            location_address = "Intersection of Picnic Way and Loiter Way, Belle Isle, Detroit, MI 48207"
+        elif "nature zoo" in location_name.lower():
+            location_address = "176 Lakeside Drive, Detroit, MI 48207"
+        else:
+            location_address = "Belle Isle, Detroit, MI 48207"
+        return {
+            'name': location_name,
+            'address': location_address,
+            'neighborhood': '',
+        }
+
+    def _parse_documents(self, response):
+        """
+        Get documents from separate list.
+        """
+        documents_dict = defaultdict(str)
+        for docItem in response.xpath('//div[contains(@id, "comp_101140")]//a'):
+            link_href = docItem.xpath('./@href').extract_first()
+            link_text = docItem.xpath('./text()').extract_first()
+            documents_dict[dateparse(link_text).date()] = link_href
+        return documents_dict
+
+    def _match_documents(self, item, response):
+        """
+        Match up the documents with the date to which they belong
+        """
+        meeting_dates = self._parse_date_and_times(item)
+        meeting_agendas = self._parse_documents(response)
+
+        if meeting_dates[0] in meeting_agendas:
+            return [{'url': self.allowed_domains[0]+meeting_agendas[meeting_dates[0]], 'note': 'Agenda'}]
+        return []
```

Comment:
Probably doesn't matter, since not paging in this spider, but instead of `+`ing allowed domains with url could do what you did `det_city_planning` spider and use `urljoin` + `response.url`. Also, you could unpack the tuple on 103 to make it a bit clearer what `meeting_dates[0]` is. 

```
def _match_documents(self, item, response):
    """
    Match up the documents with the date to which they belong
    """
    meeting_date, *_ = self._parse_date_and_times(item)
    meeting_agendas = self._parse_documents(response)

    if meeting_date in meeting_agendas:
        agenda_url = meeting_agendas[meeting_date]
        return [{'url': urljoin(response.url, agenda_url), 'note': 'Agenda'}]
    return []
```